lib/internal/process.js

```diff
@@ -65,6 +65,22 @@ function setupConfig(_source) {
     if (value === 'false') return false;
     return value;
   });
+  // Intl.v8BreakIterator() without data crashes with a fatal error, so throw instead.
+  if (global.hasOwnProperty('Intl') &&
+      Intl.hasOwnProperty('v8BreakIterator') &&
+      process.hasOwnProperty('config') &&
+      process.config.hasOwnProperty('variables') &&
+      process.config.variables.icu_small &&
+      !process.icu_data_dir) {
```

Comment:
@srl295.. #6266 landed. It included https://github.com/nodejs/node/commit/cdba9a6c02ba3418feacaefa575d02fab0eb594b. You should now to be able to grab `process.binding('config');` and check the `hasIntl`, `hasSmallICU`, and `usingICUDataDir` properties to do this check rather than looking at `process.config`